### ISSUE TITLE
Option to list existing "disks" in S3 bucket

### ISSUE THREAD

**Issue by akamensky** [archiecobbs/s3backer#131]
This is something I mentioned in #130 I think it would be great to have this as currently I can store many disks in one bucket using "prefix", but this would then require me remembering name of each one.

Off the top of my head this could be done by adding some sort of metadata file to the root of bucket that would contain list of prefixes/paths to all disks. And then querying this when needed.

**Comment by archiecobbs** [comment]
I agree this would be nice to have but it seems outside the scope of s3backer.

Leaving open for future inspiration.

**Comment by akamensky** [comment]
To elaborate - I'm building a Docker driver based off s3backer. The PoC works well, but some functionality that driver required to have is not present in s3backer. While I can achieve this by installing s3cmd and listing available "directories" in there that doesn't work for prefix based disks as it will be lots of files and I will need some parsing logic to filter through and figure out "names" of disks. 

If s3backer provides remote storage as a disk I believe it should be able to tell me which disks (prefixes etc) are present on that remote storage. 

That works even locally as you can call lsblk or a few other commands to get list of disks present on the system. But not in s3backer...

**Comment by archiecobbs** [comment]
The way #130 is implemented, if you specify a bucket name of `mybucket/foobar` then the effective prefix becomes `foobar/` (note trailing slash).

So this will create a "subdirectory" for each "disk" (i.e., prefix), and a ["CommonPrefix" query](https://stackoverflow.com/a/17579424/263801) should work, right?

**Comment by akamensky** [comment]
Sure that should work. Though I was also considering backward compatibility with prefixes prior to #130.

**Comment by archiecobbs** [comment]
OK we can leave that as a feature request but the cost/benefit is low so it's probably not going to be high priority.
